File: 1009. Complement of Base 10 Integer.js
/**
 * @param {number} n
 * @return {number}
 */
var bitwiseComplement = function(n) {
    let cnt = 0;
    if (n === 0) return 1;

    let num = n;

    while (n > 0) {
        cnt++;
        n = Math.floor(n / 2);
    }

    let maxi = (1 << cnt) - 1;
    return maxi ^ num;
};

File: 1009. Complement of Base 10 Integer.cpp
class Solution {
public:
    int bitwiseComplement(int n) {
        int cnt = 0;
        if(n==0) return 1;
        int num = n;
        while(n>0){
            cnt++;
            n /= 2;
        }
        int maxi = (1<<cnt) - 1;
        return maxi ^ num;
    }
};